### PR TITLE
wkhtmltopdf: unbreak on darwin

### PR DESCRIPTION
**Pull request by jiegec**
###### Description of changes

Unbreak wkhtmltopdf on darwin by changing dylib paths.

Fixes issue #11861.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jiegec** [comment]
Result of `nixpkgs-review pr 180669` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>odoo</li>
    <li>wkhtmltopdf</li>
  </ul>
</details>

File: pkgs/tools/graphics/wkhtmltopdf/default.nix [hunk 1]
```diff
@@ -25,6 +25,12 @@ mkDerivation rec {
     done
   '';
 
+  # rewrite library path
+  postInstall = if stdenv.isDarwin then ''
+    install_name_tool -change libwkhtmltox.0.dylib $out/lib/libwkhtmltox.0.dylib $out/bin/wkhtmltopdf
+    install_name_tool -change libwkhtmltox.0.dylib $out/lib/libwkhtmltox.0.dylib $out/bin/wkhtmltoimage
+  '' else "";
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/wkhtmltopdf/default.nix:32]:
```suggestion
  postInstall = lib.optionalString stdenv.isDarwin ''
    install_name_tool -change libwkhtmltox.0.dylib $out/lib/libwkhtmltox.0.dylib $out/bin/wkhtmltopdf
    install_name_tool -change libwkhtmltox.0.dylib $out/lib/libwkhtmltox.0.dylib $out/bin/wkhtmltoimage
  '';
```

File: pkgs/tools/graphics/wkhtmltopdf/default.nix [hunk 2]
```diff
@@ -42,6 +48,6 @@ mkDerivation rec {
     '';
     license = licenses.gpl3Plus;
     maintainers = with maintainers; [ jb55 ];
-    platforms = with platforms; linux;
+    platforms = with platforms; linux ++ darwin;
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/wkhtmltopdf/default.nix:51]:
```suggestion
    platforms = platforms.unix;
```

File: pkgs/tools/graphics/wkhtmltopdf/default.nix [hunk 3]
```diff
@@ -1,4 +1,4 @@
-{ mkDerivation, lib, fetchFromGitHub, qtwebkit, qtsvg, qtxmlpatterns
+{ stdenv, mkDerivation, lib, fetchFromGitHub, qtwebkit, qtsvg, qtxmlpatterns
 , fontconfig, freetype, libpng, zlib, libjpeg
 , openssl, libX11, libXext, libXrender }:
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/wkhtmltopdf/default.nix:4]: it should be stdenv.mkDerivation

File: pkgs/tools/graphics/wkhtmltopdf/default.nix (hunk 1 above)
Comment (jiegec) [reply pkgs/tools/graphics/wkhtmltopdf/default.nix]: Fixed

File: pkgs/tools/graphics/wkhtmltopdf/default.nix (hunk 2 above)
Comment (jiegec) [reply pkgs/tools/graphics/wkhtmltopdf/default.nix]: Fixed

File: pkgs/tools/graphics/wkhtmltopdf/default.nix (hunk 3 above)
Comment (jiegec) [reply pkgs/tools/graphics/wkhtmltopdf/default.nix:4]: Fixed

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 180669` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>odoo</li>
    <li>wkhtmltopdf</li>
  </ul>
</details>

**Comment by jiegec** [comment]
> 

This is due to qt broken, will be fixed by 180327.

**Pull request merged**